biggraphite/drivers/_downsampling.py

```diff
@@ -19,9 +19,77 @@
 
 import array
 import math
+import Queue
+import threading
+
 
 _NaN = float("NaN")
 
+_MAX_WRITE_DELAY = 5 * 60  # in seconds
+
+
+class _Writer(threading.Thread):
```

Comment:
The writer should probably not be in _downsampling.py (the role of _downsampling.py is to downsample points only). You can put that in _background_writer.py and have a self._background_writer in the accessors (or possibly have a Mixin that adds this capability)